src/rust/src/packed_r_tree.rs

```diff
@@ -587,44 +584,16 @@ impl PackedRTree {
         });
         let mut results = Vec::new();
 
-        while let Some(next) = queue.pop_front() {
-            debug!(
-                "popped node: {next:?},  remaining queue len: {}",
-                queue.len()
-            );
-            let start_node = next.nodes.start;
-            let is_leaf_node = next.level == 0;
-            if is_leaf_node {
-                assert!(start_node >= leaf_nodes_offset);
-            } else {
-                assert!(start_node < leaf_nodes_offset);
-                assert!(next.nodes.end < leaf_nodes_offset);
-            }
-            // find the end index of the nodes
-            let mut end_node = min(
```

Comment:
All this index offsetting now happens *before* the node ranges are enqueued rather than cleaning up after the fact.